File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.cs
public class Solution {
    public int solution(long num) {
        int answer = 0;
        
        if(num==1) {
            return 0;
        }
        
        for(int i=0; i<500; i++)
        {
            answer++;
            
            if(num%2==0) {
                num = num / 2;
            }
            else {
                num = num * 3 + 1;
            }
            
            if(num==1) {
                return answer;
            }
        }
        return -1;
    }
}

File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    int answer = 0;
    unsigned long long num1 = num;
    while(num1 != 1){
        if(answer == 500){
            return -1;
            break;
        }
        if(num1 % 2 == 0){
            num1 /= 2;
            answer++;
        }
        else if(num1 % 2 == 1){
            num1 *= 3;
            num1 += 1;
            answer++;
        }
    }
    return answer;
}